opentelemetry-sdk/src/trace/tracer.rs

```diff
@@ -210,18 +207,34 @@ impl opentelemetry::trace::Tracer for Tracer {
             attribute_options.truncate(span_attributes_limit);
             let dropped_attributes_count = dropped_attributes_count as u32;
 
-            let mut link_options = builder.links.take();
-            let mut links = EvictedQueue::new(span_limits.max_links_per_span);
-            if let Some(link_options) = &mut link_options {
+            // Links are available as Option<Vec<Link>> in the builder
+            // If it is None, then there are no links to process.
+            // In that case Span.Links will be default (empty Vec<Link>, 0 drop count)
+            // Otherwise, truncate Vec<Link> to keep until limits and use that in Span.Links.
+            // Store the count of excess links into Span.Links.dropped_count.
+            // There is no ability today to add Links after Span creation,
+            // but such a capability will be needed in the future
+            // once the spec for that stabilizes.
+
+            let spans_links_limit = span_limits.max_links_per_span as usize;
+            let span_links: SpanLinks = if let Some(mut links) = builder.links.take() {
+                let dropped_count = links.len().saturating_sub(spans_links_limit);
+                links.truncate(spans_links_limit);
                 let link_attributes_limit = span_limits.max_attributes_per_link as usize;
-                for link in link_options.iter_mut() {
+                for link in links.iter_mut() {
                     let dropped_attributes_count =
                         link.attributes.len().saturating_sub(link_attributes_limit);
                     link.attributes.truncate(link_attributes_limit);
                     link.dropped_attributes_count = dropped_attributes_count as u32;
                 }
-                links.append_vec(link_options);
-            }
+                SpanLinks {
+                    links,
+                    dropped_count: dropped_count as u32,
+                }
+            } else {
+                SpanLinks::default()
+            };
```

Comment:
Nit - Above logic can be part of the `From` trait for `SpanLimits` Something like:
```rust
impl From<(Vec<Link>, &SpanLimits)> for SpanLinks {
    fn from((mut links, span_limits): (Vec<Link>, &SpanLimits)) -> Self {
        ...
        ...
       SpanLinks {
            links,
            dropped_count: dropped_count as u32,
        }
    }
 }
````
 But can be as enhancement separately.

Reply:
https://github.com/open-telemetry/opentelemetry-rust/issues/1344